teps/0017-results-api.md

```diff
@@ -0,0 +1,711 @@
+add them under "/teps/images/". It's upto the TEP author to choose the name
+of the file, but general guidance is to include at least TEP number in the
+file name, for example, "/teps/images/NNNN-workflow.jpg".
+-->
+
+### Proposed Spec
+
+We propose the following API, heavily influenced by API recommendations defined
+by https://aip.dev.
+
+```proto
+syntax = "proto3";
+
+package tekton.results.v1;
+
+service Results {
+  rpc CreateResult(CreateResultRequest) returns (Result) {
+    option (google.api.http) = {
+      post: "/v1/{parent=*}/results"
+      body: "result"
+    };
+  }
+
+  rpc UpdateResult(UpdateResultRequest) returns (Result) {
+    option (google.api.http) = {
+      patch: "/v1/{result.name=*/results/*}"
+      body: "result"
+    };
+  }
+
+  rpc GetResult(GetResultRequest) returns (Result) {
+    option (google.api.http) = {
+      get: "/v1/{name=*/results/*}"
+    };
+  }
+
+  rpc DeleteResult(DeleteResultRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1/{name=*/results/*}"
+    };
+  }
+  rpc ListResults(ListResultsRequest) returns (ListResultsResponse) {
+    option (google.api.http) = {
+      get: "/v1/{parent=*}/results"
+    };
+  }
+}
+
+/***********
+ * RPC Messages
+ ***********/
+
+
+message CreateResultRequest {
+  // User provided parent to partition results under.
+  string parent = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      child_type: "tekton.results.v1.Result"
+    }];
+
+  Result result = 2 [(google.api.field_behavior) = REQUIRED];
+}
+
+message DeleteResultRequest {
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "tekton.results.v1.Result"
+    }];
+}
+
+message UpdateResultRequest {
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "tekton.results.v1.Result"
+    }];
+
+  Result result = 2 [(google.api.field_behavior) = REQUIRED];
+
+  google.protobuf.FieldMask update_mask = 3;
+  // If set, then the request will succeed if and only if the provided etag
+  // matches the server-computed value, else will fail.
+  // Should be used to prevent concurrent updates from overwritting each other.
+  string etag = 4;
+}
+
+message GetResultRequest {
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "tekton.results.v1.Result"
+    }];
+}
+
+message ListResultsRequest {
+  string parent = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      child_type: "tekton.results.v1.Result"
+    }];
+
+  string filter = 2;
+
+  int32 page_size = 3;
+  string page_token = 4;
+}
+
+message ListResultsResponse {
+  repeated Result results = 1;
+  string next_page_token = 2;
+}
+```
+
+- API Server implementations must support all methods and fields within the
```

Comment:
My advice would be to leave specifics of the API, and especially any concept of "conformance", outside the TEP.

The TEP is trying to convey "we should do X" with just enough detail about X to convince people  you have an idea how you'd do it (e.g., outline of a `Results` type and "there will be a CRUD API").

Diving into this much detail runs the risk of bikeshedding that would make the TEP take longer to land, which is a shame since I think everyone agrees we want this, and probably pretty urgently.

Once the TEP lands you can create a repo and make whatever changes you want with whatever other contributors/stakeholders you find.

Reply:
Done. I feel like I might have gone a little too far in the other direction, so let me know what areas I should add more detail for.